store/sqlstore/channel_store.go

```diff
@@ -423,6 +437,217 @@ func (s SqlChannelStore) createIndexesIfNotExists() {
 	s.CreateIndexIfNotExists("idx_channels_scheme_id", "Channels", "SchemeId")
 }
 
+// MigrateSidebarCategories creates 3 initial categories for all existing user/team pairs
+// **IMPORTANT** This function should only be called from the migration task and shouldn't be used by itself
+func (s SqlChannelStore) MigrateSidebarCategories(fromTeamId, fromUserId string) (map[string]interface{}, error) {
```

Comment:
nit: We are adding 806 lines to an already huge file (3185 LOC) for a very specific domain inside the "channel" concept, I would suggest to move all this into a `store/sqlstore/channel_categories_store.go`. Not necesarilly an independent store (but that would be an option).

Reply:
Yeah, I was kind of thinking that myself when reviewing the original PR. It definitely made it a bit of a pain when editing the file. 

That's a really good idea to just split that into its own file though. I was thinking we'd need to refactor a bunch of stuff to get that to work, but just moving it is a lot easier. I might make a followup to do that.